python/ray/tests/test_basic.py

```diff
@@ -1747,6 +1747,154 @@ def g(n):
     assert duration > 1
 
 
+def test_resource_assignment(shutdown_only):
+    '''This test makes sure that we assign resources to actors at instantiation.
+    '''
+    ray.init(num_cpus=1, num_gpus=1, resources={"Custom": 1})
+
+    @ray.remote
+    class Actor1(object):
+        def __init__(self):
+            self.resources = ray.get_resource_ids()
+
+        def get_resources(self):
+            return self.resources
+
+        def method(self):
+            return ray.get_resource_ids()
+
+        def kill(self):
+            exit(0)
+
+    # Test defaults to no actor cpus and one cpu per method.
```

Comment:
I think we can write this test a bit differently. Maybe define the `Actor` class without any decorator and then do

```python
for decorator_args in [{"num_cpus": 0.5, "num_gpus": 0.5, "resources": {"Custom1": 0.5}]:
    for instantiation_args in [{"num_cpus": 0.75, "num_gpus": 0.75, "resources": {"Custom2": 0.75}]:
        actor_class = ray.remote(**decorator_args)(Actor)
        actor = actor_class.remote(**instantiation_args)
        resources = ray.get(actor.method.get_resources.remote())
        assert ...
```

Some comments. I wouldn't expect `get_resources` and `method` to return anything different. Do they? Does `__init__` run without the method CPUs?

Also, the `kill` should be unnecessary, since the actors will go out of scope automatically if you reuse the variable name.

Reply:
I think when from `get_resources` you get the actor resources while `method` returns the actor method resources. I'm planning to change `method` to `get_actor_method_resources` and `get_resources` to `get_actor_resources`.